### ISSUE TITLE
add a defined(..) function to determine whether a context variable is defined

### ISSUE THREAD

**Issue by igor47** [taskcluster/json-e#273]
currently, if i use an unset variable in an `{$if}` then i get an error. i think an `{$if}` block should handle undefined variables like they're false. in other words, currently:

```
context: {}
template: {$if: var, then: 't', else: 'f'}
error: 'unknown context value var'
```

should instead be:

```
context: {}
template: {$if: var, then: 't', else: 'f'}
result: f
```

**Comment by djmitche** [comment]
My impression is that languages which handle missing variables by picking default values (Perl and PHP come to mind, along with old, non-strict JS's) tend to lead to difficult-to-maintain code.

JSON-e's lack of a short-circuiting logical operator (#244) makes this a little more annoying in JSON-e, since there's no way to guard an expression like `{$if: "renderType = 'bar' && someBarOnlyVariable = 10"}` -- that will cause an error even if `renderType` is not `"bar"`.  Also, we have no programmatic way check whether a variable is defined -- but adding a `defined(varName)` function shouldn't be too hard.  Then you could write

```
template: ${if: "defined('var') && var", then: ..}
```

Would that suit?

**Comment by igor47** [comment]
a `defined` function would be fine, and i appreciate explicit vs implicit handling.

**Comment by Nicrii** [comment]
@owlishDeveloper I’d like to work on this issue

**Comment by owlishDeveloper** [comment]
@Nicrii all yours!

**Comment by owlishDeveloper** [comment]
Fixed in #326
